src/content-tags-drawer/TagBubble.jsx

```diff
@@ -8,35 +8,45 @@ import PropTypes from 'prop-types';
 import TagOutlineIcon from './TagOutlineIcon';
 
 const TagBubble = ({
-  value, subTagsCount, implicit, level,
+  value, implicit, level, lineage, removeTagHandler,
 }) => {
   const className = `tag-bubble mb-2 ${implicit ? 'implicit' : ''}`;
   const tagIcon = () => (implicit ? <TagOutlineIcon className="implicit-tag-icon" /> : <Tag />);
+
+  const handleClick = (e) => {
+    if (e.target.value) {
+      e.target.checked = false;
+      removeTagHandler(e);
+    }
+  };
+
```

Comment:
Any time you create a handler function inside a React component, you should wrap it in `React.useCallback`. Otherwise, the function changes every time the component is rendered, and the child will be re-rendered unnecessarily.

All you really need here is something like this:

```js
  const handleClick = React.useCallback((e) => {
    if (!implicit) {
      removeTagHandler(lineage);
    }
  }, [implicit, lineage]);
```

Notice that you can just pass the `lineage` as an array if you want to - no need to convert it to a string first.

and then below just use 

```js
    onIconAfterClick={handleClick}
```

Reply:
This seems to have fixed the issue, thanks for the pointer!